src/main/java/org/minions/devfund/angela/Squarelotron.java

```diff
@@ -0,0 +1,128 @@
+        size = dimension;
+        squarematrix = new int[dimension][dimension];
+        int counter = 0;
+        for (int i = 0; i < dimension; i++) {
+            for (int j = 0; j < dimension; j++) {
+                counter++;
+                getSquarematrix()[i][j] = counter;
+            }
+        }
+    }
+
+    /**
+     * Constructor for {@link Squarelotron}.
+     *
+     * @param matrix the matrix to fill the squarematrix.
+     */
+    private Squarelotron(final int[][] matrix) {
+        size = matrix.length;
+        squarematrix = matrix;
+    }
+
+    /**
+     * Performs the Upside-Down Flip of the squarematrix.
+     *
+     * @param ring int ring number to perform the Upside-Down Flip.
+     * @return {@link Squarelotron}.
+     */
+    Squarelotron upsideDownFlip(final int ring) {
+        int[][] newNumbers = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                if (isPartOfTheRing(ring, i, j)) {
+                    newNumbers[size - 1 - i][j] = squarematrix[i][j];
+                } else if (ring == 0) {
+                    newNumbers[size - 1 - i][j] = squarematrix[i][j];
+                } else {
+                    newNumbers[i][j] = squarematrix[i][j];
+                }
+            }
+        }
+        return new Squarelotron(newNumbers);
+    }
+
+    /**
+     * Verifies if matrix position is part of a ring.
+     *
+     * @param ring int ring value.
+     * @param i    int x position.
+     * @param j    int y position.
+     * @return true if position is part of the ring.
+     */
+    private boolean isPartOfTheRing(int ring, int i, int j) {
+        return i == ring - 1 || i == size - ring && j >= ring - 1 && j <= size - ring
+                || j == ring - 1 || j == size - ring && i >= ring - 1 && i <= size - ring;
+    }
+
+
+    /**
+     * Performs the Main Diagonal Flip of the squarematrix.
+     *
+     * @param ring int ring number to perform the Main Diagonal Flip.
+     * @return {@link Squarelotron}.
+     */
+    Squarelotron mainDiagonalFlip(final int ring) {
+        int[][] newNumbers = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                if (isPartOfTheRing(ring, i, j)) {
+                    newNumbers[j][i] = squarematrix[i][j];
+                } else if (ring == 0) {
+                    newNumbers[j][i] = squarematrix[i][j];
+                } else {
+                    newNumbers[i][j] = squarematrix[i][j];
+                }
+            }
+        }
+        return new Squarelotron(newNumbers);
+    }
+
+    /**
+     * Rotates the squarematrix to right.
+     *
+     * @param numberOfTurns indicates the number of times the entire
+     *                      squarematrix should be rotated 90° clockwise .
+     */
+    public void rotate(final int numberOfTurns) {
+        int turnsNumber = Math.abs(numberOfTurns);
```

Comment:
You can improve this.
Instead of rotating all the time 
you can calculate the actual turn
if the numberOfTurns is 4 nothing should happen
if the numberOfTurns is 5 only it should rotate 1 time.